### PR TITLE
henry/dtra-855/fix: make sure that previous ticks stream is forgotten after changing…

### PR DESCRIPTION
**Pull request by henry-deriv**
… to another symbol in digits widget

Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by maryia-deriv — APPROVED** [review]

**Review by akmal-deriv — APPROVED** [review]

**Review by kate-deriv — APPROVED** [review]

**Review by balakrishna-deriv — DISMISSED** [review]

**Review by ahmadtaimoor-deriv — APPROVED** [review]

File: src/javascript/app/pages/trade/charts/digit_info.js [hunk 1]
```diff
@@ -249,11 +249,13 @@ const DigitInfo = (() => {
...
             if (chart.series[0].name === tick.tick.symbol) {
                 stream_id = tick.tick.id || null;
-                update(tick.tick.symbol, tick.tick.quote);
+                update(tick.tick.symbol, addComma(tick.tick.quote, decimal_places).replace(',', ''));
             } else {
                 BinarySocket.send({ forget: (tick.tick.id).toString() });
+                stream_id = null;
             }
         } else {
+            stream_id = tick.tick.id || null;
```
Comment (kate-deriv) [inline src/javascript/app/pages/trade/charts/digit_info.js:259]: `stream_is = tick.tick.id || null `and `update(tick.tick.symbol, addComma(tick.tick.quote, decimal_places).replace(',','')` are used in both conditions. Can we rework this if-else structure?

Comment (henry-deriv) [reply src/javascript/app/pages/trade/charts/digit_info.js:259]: there are three conditions

Comment (kate-deriv) [reply src/javascript/app/pages/trade/charts/digit_info.js:259]: Something like this, but 100% I missed something))))
```
    const updateChart = async (tick) => {
        const decimal_places = await Symbols.getUnderlyingPipSize(tick.tick.symbol);
        stream_id = tick.tick.id || null;
        if (stream_id && chart.series[0].name !== tick.tick.symbol) {
                BinarySocket.send({ forget: (tick.tick.id).toString() });
                stream_id = null;
        } else {
            update(tick.tick.symbol, addComma(tick.tick.quote, decimal_places).replace(',', ''));
        }
    };
```

Comment (kate-deriv) [reply src/javascript/app/pages/trade/charts/digit_info.js:259]: Please, ignore if irrelevant

Comment (henry-deriv) [reply src/javascript/app/pages/trade/charts/digit_info.js:259]: your else doesnt trigger when stream_id is null even though chart.series[0].name === tick.tick.symbol is true

**Review by kate-deriv — APPROVED** [review]

File: src/javascript/app/pages/trade/charts/digit_info.js (hunk 1 above)
Comment (maryia-deriv) [reply src/javascript/app/pages/trade/charts/digit_info.js:259]: how about this one? :)
```
            if (chart.series[0].name === tick.tick.symbol) {
                stream_id = tick.tick.id || null;
                update(tick.tick.symbol, addComma(tick.tick.quote, decimal_places).replace(',', ''));
            } else if (stream_id) {
               BinarySocket.send({ forget: (tick.tick.id).toString() });
                stream_id = null;
            }
```

**Review by maryia-deriv — CHANGES_REQUESTED** [review]

File: src/javascript/app/pages/trade/charts/digit_info.js (hunk 1 above)
Comment (henry-deriv) [reply src/javascript/app/pages/trade/charts/digit_info.js:259]: i think this one ignores the case when chart.series[0].name !== tick.tick.symbol

quite delicate the logic on this.

Essentially i'll likely just end up with if elsif and else.

Comment (henry-deriv) [reply src/javascript/app/pages/trade/charts/digit_info.js:259]: lets keep it as is. Its decently readable as of now. The flow is quite easy to understand.

Comment (maryia-deriv) [reply src/javascript/app/pages/trade/charts/digit_info.js:259]: ok, here it is :)):
```
            if (!stream_id || chart.series[0].name === tick.tick.symbol) {
                stream_id = tick.tick.id || null;
                update(tick.tick.symbol, addComma(tick.tick.quote, decimal_places).replace(',', ''));
            } else if (stream_id) {
               BinarySocket.send({ forget: (tick.tick.id).toString() });
                stream_id = null;
            }
```

Comment (henry-deriv) [reply src/javascript/app/pages/trade/charts/digit_info.js:259]: i think else if doesnt trigger when only chart.series[0].name !== tick.tick.symbol is true right

Comment (henry-deriv) [reply src/javascript/app/pages/trade/charts/digit_info.js:259]: im more confused now than before lol

Comment (maryia-deriv) [reply src/javascript/app/pages/trade/charts/digit_info.js:259]: in the original condition this part:
```
BinarySocket.send({ forget: (tick.tick.id).toString() });
stream_id = null;
```
should be triggered when `stream_id` is `true` and `chart.series[0].name === tick.tick.symbol` is `false` at the same time

**Review by balakrishna-deriv — DISMISSED** [review]

File: src/javascript/app/pages/trade/charts/digit_info.js (hunk 1 above)
Comment (akmal-deriv) [reply src/javascript/app/pages/trade/charts/digit_info.js:259]: @maryia-deriv let's not touch what has been working for many years. There's more than one way how to do stuff, programming is not an exact science)

**Review by ahmadtaimoor-deriv — APPROVED** [review]

**Review by maryia-deriv — APPROVED** [review]

**Pull request merged**